docs/premium-features/custom-types-and-actions.md

```diff
@@ -59,70 +61,134 @@ To delete a Module, follow these steps:
-- They can be used as input types (arguments) and return types for GraphQL APIs.
+  - For REST APIs, they can be used as request and response bodies.
+  - For GraphQL APIs, they can be used as input types (arguments) and return types.
+- They can define the data transferred between services as part of an event-driven architecture using a message broker (planned for future versions).
 
-Amplication provides built-in DTOs for common use cases, such as pagination, sorting, and filtering. These DTOs are automatically generated based on your application's entities. However, you can also create custom DTOs to suit your specific requirements.
+Amplication provides built-in DTOs for common use cases, such as pagination, sorting, and filtering. These DTOs are automatically generated based on your application's entities to support the CRUD APIs generated out-of-the-box. However, you can also create custom DTOs to suit your specific requirements and use them in your custom APIs.
 
 ### Differences between DTOs and Enums
 
 Enums are used to define a fixed set of constants. DTOs are used to define the structure and shape of data objects that are transferred between different parts of the project.
 
-In Amplication, you can create and manage both enums and DTOs using the Custom Types & Actions feature, allowing you to define the data contracts and constraints for your application's API endpoints.
+In Amplication, you can create and manage both enums and DTOs using the Custom Types & Actions feature, allowing you to define the data contracts and constraints for your application's API endpoints. A specific enum can also function as a DTO and can be defined as the input or output DTO for an API.
 
 ### Create DTO
 
 To create a new DTO, follow these steps:
 
+#### DTO Name and Description
+
 1. Click on the "Add DTO" button in the top right of the APIs tab.
 2. In the dropdown, click on "Add DTO".
-3. Provide a name for your DTO.
-4. Click the "Create DTO" button to finalize the creation process.
+3. Provide a name for your DTO, such as "MyNewDtoName". This will automatically populate the Name field with the provided name.
+4. Optionally, add a description for your DTO.
 
-![Create DTO](./assets/custom-types-and-actions/create-dto.png)
+#### Adding Properties
 
-### Edit, Disable, and Delete DTO
+1. In the "Properties" section, click on "Add property" to define the DTO's properties.
+2. For each property:
+   - Specify the Name of the property (e.g., "firstProperty").
+   - Choose the Type of the property from the dropdown (e.g., "String", "Number", "Boolean", "Date", "JSON", "DTO", "Enum").
+   - If needed, you can change the property to an array by toggling the "Array" switch.
+   - Mark the property as optional by checking the "Optional" checkbox.
+3. Repeat step 2 for each additional property you want to add to the DTO.
+4. If you need to define a union type for a property, click on "+ Add union type" and specify the types for the union.
```

Comment:
Let's remove the union description- as we will soon delete this code and avoid releasing this option (at this dev phase)